libs/gi/db/src/Database/DataManagers/OptConfigDataManager.ts

```diff
@@ -307,3 +308,33 @@ export function handleArtSetExclusion(
   }
   return artSetExclusion
 }
+
+function mapFilterInplace<T>(xs: T[], f: (x: T) => T | undefined): T[] {
+  const a: T[] = []
+  let reuse = true
```

Comment:
Can you predetermine `reuse` with a separate loop? The assumption is that reads are less expensive than push(writes)

Reply:
I could, but it would end up calling `f` twice for each array element.